specs/validator/tableland-openapi-spec.yaml

```diff
@@ -0,0 +1,304 @@
+openapi: 3.0.3
+info:
+  title: Tableland Validator - OpenAPI 3.0
+  description: |-
+    In Tableland, Validators are the execution unit/actors of the protocol.
+    They have the following responsibilities:
+    - Listen to on-chain events to materialize Tableland-compliant SQL queries in a database engine (currently, SQLite by default).
+    - Serve read-queries (e.g: SELECT * FROM foo_69_1) to the external world.
+    - Serve state queries (e.g. list tables, get receipts, etc) to the external world.
+
+    In the 1.0.0 release of the Tableland Validator API, we've switched to a design first approach!
+    You can now help us improve the API whether it's by making changes to the definition itself or to the code.
+    That way, with time, we can improve the API in general, and expose some of the new features in OAS3.
+
+  termsOfService: http://docs.tableland.xyz
+  contact:
+    name: core devs
+    email: carson@textile.io
+  license:
+    name: Apache 2.0
+    url: http://www.apache.org/licenses/LICENSE-2.0.html
+  version: 1.0.0
+externalDocs:
+  description: Tableland docs
+  url: http://docs.tableland.xyz
+servers:
+  - url: https://testnet.tableland.xyz/api/v1
+tags:
+  - name: query
+    description: Query the Tableland network
+    externalDocs:
+      description: Find out more about queries
+      url: http://docs.tableland.xyz
+  - name: receipt
+    description: Get information about transaction progress
+    externalDocs:
+      description: Find out more about receipts
+      url: http://docs.tableland.xyz
+  - name: tables
+    description: Access to table information
+    externalDocs:
+      description: Find out more about tables
+      url: http://docs.tableland.xyz
+paths:
+  /health:
+    get:
+      tags:
+        - health
+      summary: Returns OK if the validator considers it's healty.
+      description: Returns OK if the validator considers it's healthy.
+      operationId: health
+      responses:
+        "200":
+          description: The validator is healthy.
+  /version:
+    get:
+      tags:
+        - version
+      summary: Returns version information about the validator daemon.
+      description: Returns version information about the validator.
+      operationId: version
+      responses:
+        "200":
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/VersionInfo"
+  /query:
+    get:
+      tags:
+        - query
+      summary: Query the Tableland network
+      description: Returns the results of a SQL read query against the Tabeland network
+      operationId: queryFromQuery
+      parameters:
+        - name: statement
```

Comment:
I just realized currently, this parameter is called `s`, but we are now using `statement`. Should we stick with `s` to keep this pretty much backward compatible?

Reply:
I took care to be backwards compatible in the implementation. We can use `statement` for V1 and the old `s` in JSON-RPC.

Reply:
@carsonfarmer, a similar thing happened with the legacy `output` that we renamed to `format`.

Reply:
ah right, ok awesome. But apart from the JSON-RPC > REST switch, we also have an existing /query API, so that was the part I was most curious about.

Reply:
I just want to make sure I'm fully understanding what you're saying here. Because yes, the SDKs hide the differences between the JSON RPC API calls and the fact that they'll now be calling the REST type  /query endpoint. So that's all good. _But_, some contracts have full blown /query API strings embedded in them, and while we know they'll have to update the endpoints anyway, if the only thing they have to change is testnet -> testnets, that'll make the transition easier.

Reply:
I'd expect the `?s=` to be at the start of whatever hardcoded query exists, so it's quite near to the DNS.

I'm totally open to switch `statement` to `s`, so if that sounds better we can change it.

Reply:
naw, I'm probably being paranoid. let's leave it at statement for now and if someone else flags it, we can change